reactor-core/src/jcstress/java/reactor/core/publisher/MonoDelayStressTest.java

```diff
@@ -1,162 +1,113 @@
 import org.openjdk.jcstress.infra.results.IIII_Result;
-import reactor.core.CoreSubscriber;
+import org.openjdk.jcstress.infra.results.III_Result;
+import org.reactivestreams.Subscription;
+
+import reactor.test.scheduler.VirtualTimeScheduler;
 
 import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE;
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE_INTERESTING;
 
 public abstract class MonoDelayStressTest {
 
+	private static final int REQUEST_BEFORE_TICK                          = 0b0011_0_111; // 55
+	private static final int REQUEST_AFTER_TICK                           = 0b0010_0_111; // 39
+	private static final int CANCELLED_AFTER_REQUEST_FIRST                = 0b0111_0_001; //113
+	private static final int CANCELLED_AFTER_REQUEST_FIRST_AND_DELAY_DONE = 0b0111_0_011; //115
+	private static final int CANCELLED_AFTER_REQUEST_SECOND               = 0b0110_0_011; // 99
+	private static final int CANCELLED_BEFORE_REQUEST_BUT_DELAY_DONE      = 0b0100_0_011; // 67
+	private static final int CANCELLED_EARLY                              = 0b0100_0_001; // 65
+	private static final int CANCELLED_SUPER_EARLY                        = 0b0100_0_000; // 64
+
 	@JCStressTest
-	@Outcome(id = {"1, 1, 0, 0, 0"}, expect = ACCEPTABLE, desc = "No error dropped, composite error delivered. Cancel signal is late in that case")
-	@Outcome(id = {"1, 1, 1, 1, 0", "1, 1, 0, 1, 0"}, expect = ACCEPTABLE, desc = "Main error possibly dropped, inner or composite error delivered. Main is cancelled. Cancel signal is late")
-	@Outcome(id = {"1, 1, 1, 0, 1", "1, 1, 0, 0, 1"}, expect = ACCEPTABLE, desc = "Inner error possibly dropped, main or composite error delivered. Inner is cancelled. Cancel signal is late")
-	@Outcome(id = {"1, 0, 1, 1, 1", "1, 0, 2, 1, 1"}, expect = ACCEPTABLE, desc = "No error delivered. One composite error delivered or both errors are dropped. Cancel signal is propagated to both")
+	@Outcome(id = {"1, 0, " + REQUEST_BEFORE_TICK}, expect = ACCEPTABLE, desc = "Request before tick was delivered")
+	@Outcome(id = {"1, 0, " + REQUEST_AFTER_TICK}, expect = ACCEPTABLE, desc = "Request AFTER tick was delivered")
 	@State
-	public static class InnerOnErrorAndOuterOnErrorAndCancelStressTest {
+	public static class RequestAndRunStressTest {
 
-		final StressSubscriber<Integer> subscriber = new StressSubscriber<Integer>(1L);
+		/*
+		Implementation notes: in this test we use the VirtualTimeScheduler to better coordinate
+		the triggering of the `run` method. We directly interpret the end STATE to track what
+		happened.
+		 */
 
-		StressSubscription<Integer> subscriptionOuter;
-		StressSubscription<Integer> subscriptionInner;
+		final StressSubscriber<Long> subscriber = new StressSubscriber<>(0L);
+		final VirtualTimeScheduler   virtualTimeScheduler;
+		final MonoDelay              monoDelay;
+		final AtomicReference<MonoDelay.MonoDelayRunnable> subscriptionRef = new AtomicReference<>();
```

Comment:
```suggestion

		MonoDelay.MonoDelayRunnable> subscription;
```

AtomicRef is redundant and it can be simplified to just field since actors will be run after the obj is initialized so we have HB